### DISCUSSION TITLE
Teleport auto-users not automatically creating users?

### DISCUSSION THREAD

**Discussion by dcarriger** [gravitational/teleport#21269] [Q&A]
I followed the steps at https://goteleport.com/docs/server-access/guides/host-user-creation/ to enable automatic user creation. Initially, it didn't work, until I realized that I hadn't logged out and logged back in, so my roles hadn't refreshed. Oops, that was in the docs, missed it. However, now that I've sorted that out...

```
tsh ssh my-servername-here
Failed to launch: user: unknown user dcarriger.
ERROR: Process exited with status 255
```

The docs heavily imply that Teleport is going to do a bunch of `useradd` commands under the hood to set my account up. Teleport is running as root and all of the shell commands listed are in root's $PATH. I've confirmed that the `/etc/teleport.yaml` on the remote host isn't denying host user creation. All of the roles assigned to my user allow for host user creation. What am I missing here?

**Comment by stevenGravy** [comment]
@dcarriger please provide what linux flavor you are doing this on. Are there any logs on the server side?  Please share your configuration and role settings.

**Comment by dcarriger** [comment]
@stevenGravy I went over everything again and found an extra role had been attached to my user that I wasn't expecting, and that was missing the required `create_host_user` option. So, completely user error. However, the logs were not helpful whatsoever from the Teleport auth/proxy side and gave no indication of anything like "user not found, not creating due to policy on XYZ". Are there any alternate logs I could have checked that would have helped me pinpoint the issue sooner?

**Comment by gdubicki** [comment]
I adapted [the guide](https://goteleport.com/docs/server-access/guides/host-user-creation/) to Terraform and it turned out that I have to set **both**:

* `create_host_user_mode`
* `create_host_user`,

...in my role code explicitly. See https://github.com/gravitational/teleport/issues/33288 for more info.
